hashintel/hash
# Add Input styles

## 🌟 What is the purpose of this PR?



Adds styling for inputs (TextField, Radio Button, Checkbox, Switches) and addresses a follow up from #409 

## ⚠️ Known issues



## 🚫 Blocked by





## 🐾 Next steps



## 🔍 What does this change?



- Override default styling for the MUI's Checkbox, Radio and Switch component to match our designs
- Adds a custom TextField component with the following props
    - `success` -> For valid state UI
    - `autoResize` -> Passing `multiline` to TextField makes it render a textarea instead of an input. When autoResize is passed in this scenario, the TextField (which is a textarea at this point) becomes resizeable
    - `showLabelCornerHint` -> Shows a "Optional" or "Required" hint at the right hand side of the label depending on what the `required` prop is.
    - `size` -> Can either be "small", "medium" or "large"
- Move inputs to `shared/ui` folder
- Adds `FormInline` component to render a TextField and Button component inline. This makes it possible to achieve this [design](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-Editor?node-id=823%3A67158)
- Stop storing hover/focus in state, but instead leverage css to handle styles. This addresses a [comment](https://github.com/hashintel/hash/pull/409#discussion_r836608704) made by @benwerner01  in #409 


### 📜 Does this require a change to the docs?



- No

## 🔗 Related links





- [Add styles for input fields](https://app.asana.com/0/1201927504137297/1201916223398334) _(internal)_

## 🛡 What tests cover this?



- None

## ❓ How to test this?



1.  Checkout the branch
1.  Confirm the different inputs in the page matches the [designs](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-Editor?node-id=823%3A67158)

## 📹 Demo


![Screenshot 2022-04-01 at 15 17 29](https://user-images.githubusercontent.com/16928688/161282042-38b87948-6e66-41fa-8993-9027ac78450e.png)



---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201916223398334
